### ISSUE TITLE
Upgrade to 8.6 is not migrating persisted agent application state from pre-8.6

### ISSUE THREAD

**Issue by blakerouse** [elastic/elastic-agent#2050]
Elastic Agent 8.6 and the new V2 architecture changes how runtime directories are used for state. This had to be done because of the new design runs each input type in its own process so each input type needs its own state directory. In pre-8.6 it was all shared in the same directory per beat type.

This means that on an upgraded 8.6 from a previous pre-8.6 version of Elastic Agent the state directory information will be lost and cause the the spawned beats to not resume from their previous point in time but instead start from the beginning again.

This is not a problem for service runtime components because they manage their own state and are installed into there own directories this state tracking is managed by them and not by the Elastic Agent.

I believe the proper fix is to adjust the command runtime for command components to copy previous state into the work directory of the spawned process. We also need to confirm that a spawned beat uses the processes spawned work directory for state when the command line arguments for state path are not provided as that was also removed.

For confirmed bugs, please report:
- Version: 8.6
- Operating System: All + Containers

**Comment by cmacknz** [comment]
> We also need to confirm that a spawned beat uses the processes spawned work directory for state when the command line arguments for state path are not provided as that was also removed.

On my installed 8.6 agent the Filebeat registry is being created in the run directory for the Filebeat process started to run the `log-default` input:

```
sudo tree /Library/Elastic/Agent/data/elastic-agent-123730/run/log-default/

/Library/Elastic/Agent/data/elastic-agent-123730/run/log-default/
├── meta.json
└── registry
    └── filebeat
        ├── log.json
        └── meta.json

2 directories, 3 files
```

I can see the same is true for `system/metrics-default` although the `/` character in the name is creating a `metrics-default` sub-directory:

```
❯ sudo tree /Library/Elastic/Agent/data/elastic-agent-123730/run/system/

/Library/Elastic/Agent/data/elastic-agent-123730/run/system/
└── metrics-default
    └── meta.json

1 directory, 1 file
```

**Comment by cmacknz** [comment]
I've confirmed that we didn't handle this properly versions prior to 8.6, so this isn't a blocker for the 8.6 release and the underlying problem causing duplication has been fixed with the changes in 8.6. Releases from 8.6 onward will not duplicate log data.

To this this I installed an 8.3.3 agent with the custom logs integration installed, and pointed it to an arbitrary test file on my host system. This is the relevant section of the agent policy configuration the custom logs input:

```yaml
  - id: logfile-logs-6d027aef-6b75-4dcf-bc50-7e778507c2c0
    name: custom-logs-1
    revision: 1
    type: logfile
    use_output: default
    meta:
      package:
        name: log
        version: 1.1.0
    data_stream:
      namespace: default
    package_policy_id: 6d027aef-6b75-4dcf-bc50-7e778507c2c0
    streams:
      - id: logfile-log.log-6d027aef-6b75-4dcf-bc50-7e778507c2c0
        data_stream:
          dataset: generic
        paths:
          - /var/log/test-logfile.txt
```

I set it up to consume a static log file with the following arbitrary content:
```
this is a test log file
this log file should not be duplicated
is this log file duplicated?

```

I then upgrade the agent to version 8.5.3, and observed in Kibana that the entire file was sent a second time. Interestingly I observed that the file wasn't resent in its entirety until I changed by adding several new lines so that its content was:

```
this is a test log file
this log file should not be duplicated
is this log file duplicated?



this log line was added in 8.5.3
this is also an 8.5.3 log line
```

The results I got in Kibana are below and confirm that data is resent on upgrade in versions before 8.6:

<img width="2219" alt="Screen Shot 2023-01-03 at 4 14 08 PM" src="https://user-images.githubusercontent.com/3466215/210442487-371d3efa-87ef-400f-951c-c1fd91b7eb50.png">

**Comment by cmacknz** [comment]
Closed. We won't address this for the upgrade from versions prior to 8.6 to version 8.6, but this problem is fixed in 8.6 itself already with the changes to the agent directory structure.

**Comment by prachigupta-qasource** [comment]
### `Bug Conversion`

-  **01 new testcase** is created under the 'Fleet Test Suite' at:
   - https://elastic.testrail.io/index.php?/cases/view/168588

Thanks!
